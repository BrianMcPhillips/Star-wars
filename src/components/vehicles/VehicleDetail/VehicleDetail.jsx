import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const VehicleDetail = ({ 
  name, 
  model, 
  manufacturer,
  cost_in_credits,
  length,
  max_atmosphering_speed,
  crew,
  passengers,
  vehicle_class,
  pilots,
  films,
  url 
}) => (
  <div>
    <h1>{name}</h1>
    <h3>{model}</h3>
    <h4>{manufacturer}</h4>
    <h4>Cost: {cost_in_credits}</h4>
    <h4>Length: {length}</h4>
    <h4>Speed: {max_atmosphering_speed}</h4>
    <h4>Crew: {crew}</h4>
    <h4>Passengers: {passengers}</h4>
    <h4>Vehicle Class: {vehicle_class}</h4>
    <h4>Pilots: {pilots.map((pilot, i) =>
      <Link 
        to={`/char-detail/${pilot.split('people/')[1]}`}
        key={pilot + i}>
        <p>{pilot}</p>
      </Link>
    )}</h4>
    <h4>Films: {films.map((film, i) => 
      <Link
        to={`/film-detail/${film.split('films/')[1]}`}
        key={film + i}>
        <p>{film}</p>
      </Link>
    )}</h4>
  </div>
);

VehicleDetail.propTypes = {
  name: PropTypes.string.isRequired, 
  model: PropTypes.string.isRequired, 
  manufacturer: PropTypes.string.isRequired,
  cost_in_credits: PropTypes.string.isRequired,
  length: PropTypes.string.isRequired,
  max_atmosphering_speed: PropTypes.string.isRequired,
  crew: PropTypes.string.isRequired,
  passengers: PropTypes.string.isRequired,
  vehicle_class: PropTypes.string.isRequired,
  pilots: PropTypes.arrayOf(PropTypes.string),
  films: PropTypes.arrayOf(PropTypes.string),
  url: PropTypes.string.isRequired
};

export default VehicleDetail;
