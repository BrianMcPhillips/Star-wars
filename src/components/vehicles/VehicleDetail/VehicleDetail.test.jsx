import React from 'react';
import { render, cleanup } from '@testing-library/react';
import VehicleDetail from './VehicleDetail';
import { MemoryRouter } from 'react-router-dom';

describe('VehicleDetail component', () => {
  afterEach(() => cleanup());
  it('renders VehicleDetail', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <VehicleDetail 
          name="Sand Crawler" 
          model="Digger Crawler" 
          manufacturer="Corellia Mining Corporation"
          cost_in_credits="149999" 
          length="12.5"
          max_atmosphering_speed="1050"
          crew="1"
          passengers="0"
          vehicle_class="Wheeled"
          pilots= {[
            'https://swapi.dev/api/people/1/', 
            'https://swapi.dev/api/people/2/', 
            'https://swapi.dev/api/people/3/', 
            'https://swapi.dev/api/people/4/', 
            'https://swapi.dev/api/people/5/'
          ]}
          films= {[
            'https://swapi.dev/api/films/1/', 
            'https://swapi.dev/api/films/2/', 
            'https://swapi.dev/api/films/3/'
          ]} 
          url="https://swapi.dev/api/vehicles/4"
        />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
