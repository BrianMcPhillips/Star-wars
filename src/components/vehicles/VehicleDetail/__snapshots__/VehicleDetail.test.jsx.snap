// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VehicleDetail component renders VehicleDetail 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Sand Crawler
    </h1>
    <h3>
      Digger Crawler
    </h3>
    <h4>
      Corellia Mining Corporation
    </h4>
    <h4>
      Cost: 149999
    </h4>
    <h4>
      Length: 12.5
    </h4>
    <h4>
      Speed: 1050
    </h4>
    <h4>
      Crew: 1
    </h4>
    <h4>
      Passengers: 0
    </h4>
    <h4>
      Vehicle Class: Wheeled
    </h4>
    <h4>
      Pilots: 
      <a
        href="/char-detail/1/"
      >
        <p>
          https://swapi.dev/api/people/1/
        </p>
      </a>
      <a
        href="/char-detail/2/"
      >
        <p>
          https://swapi.dev/api/people/2/
        </p>
      </a>
      <a
        href="/char-detail/3/"
      >
        <p>
          https://swapi.dev/api/people/3/
        </p>
      </a>
      <a
        href="/char-detail/4/"
      >
        <p>
          https://swapi.dev/api/people/4/
        </p>
      </a>
      <a
        href="/char-detail/5/"
      >
        <p>
          https://swapi.dev/api/people/5/
        </p>
      </a>
    </h4>
    <h4>
      Films: 
      <a
        href="/film-detail/1/"
      >
        <p>
          https://swapi.dev/api/films/1/
        </p>
      </a>
      <a
        href="/film-detail/2/"
      >
        <p>
          https://swapi.dev/api/films/2/
        </p>
      </a>
      <a
        href="/film-detail/3/"
      >
        <p>
          https://swapi.dev/api/films/3/
        </p>
      </a>
    </h4>
  </div>
</DocumentFragment>
`;
